cl/users/views.py

```diff
@@ -655,3 +687,54 @@ def moosend_webhook(request: HttpRequest) -> HttpResponse:
     # Moosend does a GET when you create/edit the automation workflow,
     # so we need to return a 200 even for GETs.
     return HttpResponse("<h1>200: OK</h1>")
+
+
+@login_required
+@never_cache
+def view_webhooks(request: AuthenticatedHttpRequest) -> HttpResponse:
+    if request.method == "POST":
+        instance = Webhook()
+        form = WebhookForm(request.POST, instance=instance)
+        if form.is_valid():
+            instance.user = request.user
+            form.save()
+            return HttpResponseRedirect(reverse("view_api"))
+    else:
+        form = WebhookForm()
+
+    return render(
+        request,
+        "profile/webhooks.html",
+        {"webhook_form": form, "private": True, "page": "webhooks"},
+    )
+
+
+@login_required
+@require_http_methods(["POST"])
+def toggle_recap_email_auto_subscription(
```

Comment:
Toggles like this can be a bit dangerous, but I guess it's fine for now. Just imagine that a user:

1. Wants to disable auto-subscribe, but is busy.
2. Opens a tab, browses to the auto-subscribe page.
3. Gets distracted
4. Days later....remembers they wanted to disable auto-subscribe.
5. Opens a tab, browses to the auto-subscribe page; disables it.
6. Days later...
7. The user is closing tabs and sees the tab they opened in step 1.
8. Auto-subscribe shows it's still enabled, so they click it.
9. On the backend, it's actually disabled from step 5, so this actually *enables* auto-subscribe again. 
10. There's a message that says "Auto-subscribe is enabled"
11. The DB has it as enabled.
12. The toggle shows it as disabled.
13. Oh no.

It's a bit of an edge case, but it's the kind of thing that gets me all the time because I'm often multi-tasking. We don't have to change it, but we should try to keep this in mind in the future.

Reply:
Oh! yeah, I could reproduce it and this was a problem!!

I tried to solve it as follows:

Sending within the POST request the toggle current status and based on that status we update the DB into the view.  So it doesn't matter the status of this field in the DB, It'll be updated based on the action that the user performs when clicking the toggle.

To accomplish this we send via htmx the current status that is stored into an input field named `current_toggle_status`, we also update this status every time the user clicks the toggle via JS.

Let me know what do you think about this approach.